src/Form/Form.php

```diff
@@ -139,7 +182,7 @@ protected function _buildValidator(Validator $validator)
      */
     public function validate(array $data)
     {
-        $validator = $this->validator();
+        $validator = $this->getValidator();
```

Comment:
Are we allowed to use the non-deprecated new one inside 3.x?
I mean if someone extended validator(), would that be an issue because it is not called anymore?

Reply:
There won't be any issue.

Reply:
What do you mean?
I just remember that we had some issues with inheritance here, because people implemented the old method, and the core itself now only calls new method, thus the extended method and its functionality is not used/called anymore.
That sounds like a BC break.

Reply:
I am pretty sure we don't provide any guarantees about overridden methods in minor releases. For e.g. when `Form::input()` was deprecated in favor of `Form::control()`, bootstrap-ui plugin's `FormHelper` which had overridden `Form::input()` had to be updated.

Reply:
We only guarantee backwards compatibility with public API, not the internals.

Reply:
What is the definition of public api?
Because `public function validator(Validator ...)` sounds rather public, and might very well be called from the outside as well.

Reply:
`ValidatorAwareTrait::validator()` also uses it's new `getValidator()` and `setValidator()` internally. I am sure there are other such examples where the deprecated methods use the newly added methods internally.

Reply:
I see. So we have a more global BC problem here, strictly speaking. But maybe we can ignore it.